openmmtools/storage/iodrivers.py

```diff
@@ -0,0 +1,1653 @@
+                                                                     chunksizes=(appendable_chunk_size, 1))
+            # Specify a way for the IO Driver stores data
+            self.add_metadata('IODriver_Type', self.dtype_string())
+            self.add_metadata('type', typename(self.dtype))
+            self._unit = 'NoneType'
+            self.add_metadata('IODriver_Unit', self._unit)
+            # Specify the type of storage object this should tie to
+            self.add_metadata('IODriver_Storage_Type', self.storage_type)
+            self.add_metadata('IODriver_Appendable', 1)
+        self._dump_metadata_buffer()
+        # Set the output mode by calling the variable
+        self._output_mode
+        return
+
+    def read(self):
+        if self._bound_target is None:
+            self._bind_read()
+        # Set the output mode by calling the variable
+        self._output_mode
+        return self._decoder(self._bound_target)
+
+    def write(self, data):
+        # Check type
+        if type(data) is not self.dtype:
+            raise TypeError("Invalid data type on variable {}.".format(self._target))
+        # Bind
+        if self._bound_target is None:
+            self._bind_write(data)
+        # Check writeable
+        if self._output_mode != 'w':
+            raise TypeError("{} at {} was saved as appendable data! Cannot overwrite, must use append()".format(
+                self.dtype_string(), self._target)
+            )
+        # Save data
+        self._bound_target[:] = self._encoder(data)
+        return
+
+    def append(self, data):
+        # Check type
+        if type(data) is not self.dtype:
+            raise TypeError("Invalid data type on variable {}.".format(self._target))
+        # Bind
+        if self._bound_target is None:
+            self._bind_append(data)
+        # Check writeable
+        if self._output_mode != 'a':
+            raise TypeError("{} at {} was saved as appendable data! Cannot overwrite, must use append()".format(
+                self.dtype_string(), self._target)
+            )
+        # Determine current current length and therefore the last index
+        length = self._bound_target.shape[0]
+        # Save data
+        self._bound_target[length, :] = self._encoder(data)
+
+    @property
+    def storage_type(self):
+        return 'variables'
+
+    @property
+    def _get_on_disk_dtype(self):
+        """Function to process None for _on_disk_dtype"""
+        if self._on_disk_dtype is None:
+            return_type = self.dtype
+        else:
+            return_type = self._on_disk_dtype
+        return return_type
+
+    @property
+    def _on_disk_dtype(self):
+        """
+        Allow overwriting the dtype for storage for extending this method to cast data as a different type on disk
+        This is the property to overwrite the cast dtype
+        """
+        return None
```

Comment:
If you just return `self.dtype` instead of None, you can remove the hybrid property/getter `_get_on_disk_dtype` and you would still be able to overwrite the behavior.